plugin/src/main/scala/sbtembeddedcassandra/EmbeddedCassandraPlugin.scala

```diff
@@ -44,35 +43,56 @@ object EmbeddedCassandraPlugin extends AutoPlugin {
   lazy val embeddedCassandraDefaultSettings: Seq[Def.Setting[_]] = Seq(
     embeddedCassandraPropertiesSetting := defaultProperties,
     embeddedCassandraConfigFileSetting := defaultConfigFile,
-    embeddedCassandraWorkingDirectorySetting := new File(defaultWorkingDirectory)
+    embeddedCassandraWorkingDirectorySetting := file(defaultWorkingDirectory)
   ) ++ embeddedCassandraKeysDef
 
   lazy val embeddedCassandraKeysDef: Seq[Def.Setting[_]] = Seq(
-    embeddedCassandraCreateConfigFile := {
-      val workingDir: java.io.File       = embeddedCassandraWorkingDirectorySetting.value
-      val variables: Map[String, String] = embeddedCassandraPropertiesSetting.value + ("workingDirectory" -> workingDir.getAbsolutePath)
-
-      IOUtils
-        .copyFile(workingDir, cassandraConfInput, variables, cassandraConfOutput)
-        .logErrorOr(sbtLogger(streams.value.log), s"File created in ${workingDir.getAbsolutePath}")
-    },
     embeddedCassandraStart := {
-      val workingDir: java.io.File       = embeddedCassandraWorkingDirectorySetting.value
-      val variables: Map[String, String] = embeddedCassandraPropertiesSetting.value + ("workingDirectory" -> workingDir.getAbsolutePath)
-      val logger: Logger                 = streams.value.log
+      val workingDir: File                = embeddedCassandraWorkingDirectorySetting.value
+      val properties: Map[String, String] = embeddedCassandraPropertiesSetting.value
+      val variables: Map[String, String]  = properties + ("workingDirectory" -> workingDir.getAbsolutePath)
+      val statementsFile: Option[File]    = embeddedCassandraCQLFileSetting.value
+      val logger: Logger                  = streams.value.log
 
       import CassandraUtils._
+      import IOUtils._
 
       (for {
-        _    <- IOUtils.deleteDir(workingDir)
-        yaml <- IOUtils.copyFile(workingDir, cassandraConfInput, variables, cassandraConfOutput)
-        _    <- IOUtils.copyFile(workingDir, cassandraLog4jInput, Map.empty, cassandraLog4jOutput)
+        _    <- verifyProperties(properties)
+        _    <- deleteDir(workingDir)
+        yaml <- copyFile(workingDir, cassandraConfInput, variables, cassandraConfOutput)
+        _    <- copyFile(workingDir, cassandraLog4jInput, Map.empty, cassandraLog4jOutput)
         _    <- setCassandraProperties(yaml, workingDir, cassandraConfOutput, cassandraLog4jOutput)
         _    <- startCassandra(yaml, workingDir, 60.seconds)
+        _ <- statementsFile match {
```

Comment:
Maybe we could extract it up.

`statementsFile map { sf => for {  } yield ... } getOrElse(Right((): Unit))` ?